plugins/@grouparoo/google-sheets/src/lib/shared/Spreadsheet.ts

```diff
@@ -0,0 +1,315 @@
+import {
+  objectCache,
+  objectCacheInvalidate,
+  SimpleAppOptions,
+} from "@grouparoo/core";
+
+const { GoogleSpreadsheet } = require("google-spreadsheet");
+
+const GOOGLE_SHEETS_ROWS_LIMIT = 10000;
+const MINUTE_IN_MILLIS = 60 * 1000;
```

Comment:
maybe wait less if the rate limit is per minute. 20 seconds? it's fine to try twice and not get it.